main/CommandHandler.cpp

```diff
@@ -837,6 +837,130 @@ int getIdxChannel(const uint8_t command[], uint8_t response[])
   return 6;
 }
 
+int setEnt(const uint8_t command[], uint8_t response[])
+{
+  const uint8_t* commandPtr = &command[3];
+  uint8_t eapType;
+  char ssid[32 + 1];
+
+  memset(ssid, 0x00, sizeof(ssid));
+
+  // EAP Type - length
+  uint16_t eapTypeLen = (commandPtr[0] << 8) | commandPtr[1];
+  commandPtr += sizeof(eapTypeLen);
+
+  // EAP Type - data
+  memcpy(&eapType, commandPtr, sizeof(eapType));
+  commandPtr += sizeof(eapType);
+
+  // SSID - length
+  uint16_t ssidLen = (commandPtr[0] << 8) | commandPtr[1];
+  commandPtr += sizeof(ssidLen);
+
+  // SSID - data
+  memcpy(ssid, commandPtr, ssidLen);
+  commandPtr += ssidLen;
+
+  if (eapType == 0) {
+    // PEAP/MSCHAPv2
+    char username[128 + 1];
+    char password[128 + 1];
+    char identity[128 + 1];
+    const char* rootCA;
+    
+    memset(username, 0x00, sizeof(username));
+    memset(password, 0x00, sizeof(password));
+    memset(identity, 0x00, sizeof(identity));
+
+    // username - length
+    uint16_t usernameLen = (commandPtr[0] << 8) | commandPtr[1];
+    commandPtr += sizeof(usernameLen);
+
+    // username - data
+    memcpy(username, commandPtr, usernameLen);
+    commandPtr += usernameLen;
+
+    // password - length
+    uint16_t passwordLen = (commandPtr[0] << 8) | commandPtr[1];
+    commandPtr += sizeof(passwordLen);
+
+    // password - data
+    memcpy(password, commandPtr, passwordLen);
+    commandPtr += passwordLen;
+
+    // identity - length
+    uint16_t identityLen = (commandPtr[0] << 8) | commandPtr[1];
+    commandPtr += sizeof(identityLen);
+
+    // identity - data
+    memcpy(identity, commandPtr, identityLen);
+    commandPtr += identityLen;
+
+    // rootCA - length
+    uint16_t rootCALen = (commandPtr[0] << 8) | commandPtr[1];
+    memcpy(&rootCALen, commandPtr, sizeof(rootCALen));
+    commandPtr += sizeof(rootCALen);
+
+    // rootCA - data
+    rootCA = (const char*)commandPtr;
+    commandPtr += rootCALen;
+
+    WiFi.beginEnterprise(ssid, username, password, identity, rootCA);
+  } else {
+    // EAP-TLS
```

Comment:
@facchinm unless someone can test this path properly, I would suggest nuking this code path for now :)

Reply:
I've managed to test it :)